activesupport/lib/active_support/cache/mem_cache_store.rb

```diff
@@ -306,7 +306,14 @@ def normalize_key(key, options)
           if key
             key = key.dup.force_encoding(Encoding::ASCII_8BIT)
             key = key.gsub(ESCAPE_KEY_CHARS) { |match| "%#{match.getbyte(0).to_s(16).upcase}" }
-            key = "#{key[0, 212]}:hash:#{ActiveSupport::Digest.hexdigest(key)}" if key.size > 250
+            key_max_size = 250
+
+            if key.size > key_max_size
+              key_separator = ":hash:"
+              key_hash = ActiveSupport::Digest.hexdigest(key)
+              key_trim_size = key_max_size - key_separator.size - key_hash.size
+              key = "#{key[0, key_trim_size]}#{key_separator}#{key_hash}"
```

Comment:
Similarly here, if the key contain multi-byte characters, we might be off. We could use `byteslice`, but it may cause some strings to be invalid.